pennylane/pauli/utils.py

```diff
@@ -140,24 +140,20 @@ def are_identical_pauli_words(pauli_1, pauli_2):
     # pylint: disable=isinstance-second-argument-not-valid-type
     """Performs a check if two Pauli words have the same ``wires`` and ``name`` attributes.
 
-    This is a convenience function that checks if two given :class:`~.Tensor` instances specify the same
-    Pauli word. This function only checks if both :class:`~.Tensor` instances have the same wires and name
-    attributes, and hence won't perform any simplification to identify if the two Pauli words are
-    algebraically equivalent. For instance, this function will not identify
-    that ``PauliX(0) @ PauliX(0) = Identity(0)``, or ``PauliX(0) @ Identity(1)
-    = PauliX(0)``, or ``Identity(0) = Identity(1)``, etc.
```

Comment:
This is no longer True since the pauli rep is simplified, as pointed out by @Qottmann

Reply:
Maybe we could rename this to `are_proportional_pauli_words` because we also have
```python3
op1 = 0.5 * X(0) @ I(0)
op2 = X(0)
qml.pauli.are_identical_pauli_words(op1, op2)
```
```
True
```

Reply:
I see your point. Although (correct me if I'm wrong), when talking about Pauli words, aren't scalar factors not relevant?

Reply:
I'm not sure there is really a standard everyone agrees on, though "are identical" could defo be misleading here. Would it be a big breaking change to change it? Like are there many occurences in the codebase?

Reply:
It is used by `group_observables`, `qml.Hamiltonian`, `qml.pauli.pauli_mult`, and `qml.pauli.diagonalize_qwc_pauli_words`. Pretty contained so I definitely don't think this would be a breaking change.

Reply:
If you feel comfrotable doing the change go for it, if it is becoming a blocker it is not essential and you can leave it 👍

Reply:
I think I'd rather leave it. I see Pauli words as defined in PennyLane to not have coefficients, so "identical" Pauli words should be stripped of coefficients in my opinion. That being said, this is not something that needed discussion in the past since `Tensor` never had coefficients, so maybe it's worth re-evaluating how we want to define Pauli words. I think it still makes sense to ignore coefficients as they're introduced by Pauli sentences due to being defined as linear combinations of Pauli words.